version: '3.8'

services:
  web:
    build: ./project
    command: python app/main.py #uvicorn main:app_fastapi --app-dir=app --workers 1 --host 0.0.0.0 --port 8000 --log-config ./app/config_log.ini
    volumes:
      - ./project:/usr/src/app
    ports:
      - "8004:8000"
    environment:
      - DATABASE_URL=postgresql+asyncpg://postgres:postgres@db:5432/dev
      - TEST_DATABASE_URL=postgresql+psycopg2://postgres:postgres@db:5432/test
    depends_on:
      - db
  db:
    image: postgres:13.4
    volumes:
      - dev-db-data:/var/lib/postgresql/data
      - ./sql/create_tables.sql:/docker-entrypoint-initdb.d/create_tables.sql
      - ./sql/mock_products.csv:/docker-entrypoint-initdb.d/mock_products.csv
      - ./sql/multiple_db.sh:/docker-entrypoint-initdb.d/multiple_db.sh
    ports:
      - "6543:5432"
    restart: always
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U $$POSTGRES_USER -d $$POSTGRES_DB" ]
      interval: 5s
      timeout: 5s
      retries: 5
    environment:
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
      - POSTGRES_MULTIPLE_DATABASES=dev,test
volumes:
  dev-db-data:
